### ISSUE TITLE
Alpine build of Datadog.Trace.ClrProfiler.Native

### ISSUE THREAD

**Issue by chris-cummings** [DataDog/dd-trace-dotnet#319]
**Is your feature request related to a problem? Please describe.**
Datadog.Trace.ClrProfiler.Native is built explicitly for Linux x64 and does not work in alpine containers.

**Describe the solution you'd like**
An alpine build of Datadog.Trace.ClrProfiler.Native so it will work in .Net Core alpine containers.

**Comment by lucaspimentel** [comment]
Hey, @chris-cummings. Sorry for the delayed response. We finally go around to testing the profiler on different Linux environment and found out the following:

- works out of the box on Ubuntu 18.04 (`mcr.microsoft.com/dotnet/core/runtime:2.1-bionic`)
- works out of the box on Debian 9 (`mcr.microsoft.com/dotnet/core/runtime:2.1-stretch-slim`)
- on Alpine 3.7 (`mcr.microsoft.com/dotnet/core/runtime:2.1-alpine3.7`), requires adding the `libc6-compat` package using `apk add libc6-compat` as [documented](https://docs.datadoghq.com/tracing/setup/dotnet/?tab=netcoreonlinux)
on Alpine 3.9 (`mcr.microsoft.com/dotnet/core/runtime:2.1-alpine3.9`), *the profiler does not load*, even after adding the `libc6-compat` package

We will investigate that last one (Alpine 3.9 and leave this issue open until it is resolved.

Thanks again for your feedback!

**Comment by franck-schmidlin** [comment]
just been faffing around this.
Works well with microsoft/dotnet:2.2-sdk-alpine
but not with mcr.microsoft.com/dotnet/core/aspnet:2.2-alpine3.9 

I'm still hoping to figre out why

**Comment by chris-cummings** [comment]
Thanks guys, looks like the piece we were missing was the libc6-compat package. I will stay tuned for what you learn in from your alpine 3.9 testing.

**Comment by franck-schmidlin** [comment]
Works here with mcr.microsoft.com/dotnet/core/aspnet:2.2-alpine3.8 (+libc6-compat)
Issue seems to be specific to 3.9

**Comment by zacharycmontoya** [comment]
It looks like on Alpine 3.9 we're missing the shared library `ld-linux-x86-64.so.2`, which can be found in the `gcompat` package. I've tested locally that adding this package allows our profiler to be launched on Alpine 3.9 and it successfully performs automatic instrumentation.

@franck-schmidlin can you verify that installing `gcompat` on top of mcr.microsoft.com/dotnet/core/aspnet:2.2-alpine3.9 succeeds?

**Comment by lucaspimentel** [comment]
Related: https://github.com/DataDog/dd-trace-dotnet/issues/561

**Comment by d47id** [comment]
All the suggestions in this thread involve adding packages to enable glibc support on alpine. What the original requester is asking for is a copy of Datadog.Trace.ClrProfiler.Native that's compiled against musl-libc so it works out of the box on alpine. 

We used libc6-compat on alpine 3.7 successfully for some time, but the last supported .net core version with an alpine 3.7 base is 2.2.7. 

We attempted to use both gcompat and https://github.com/sgerrand/alpine-pkg-glibc on alpine 3.11. Although this allowed some of our services to start, others still failed and we saw random core dumps until we finally gave up and started using Debian-based runtime images.

For now, I think you guys should explicitly say you DO NOT SUPPORT Alpine, and I wouldn't consider this resolved until you provide a copy of Datadog.Trace.ClrProfiler.Native that was compiled on Alpine against musl-libc.

**Comment by seanamosw** [comment]
I believe I'm bumping into this as well.

Using the Microsoft image: mcr.microsoft.com/dotnet/core/aspnet:3.1.1-alpine3.10
Installing the apks: ibc6-compat, gcompat

The automatic instrumentation fails to start:
```
/var/log/datadog # cat dotnet-profiler.log 
[dotnet] 1: [info] Datadog CLR Profiler 1.12.0
[dotnet] 1: [debug] Initialize
[dotnet] 1: [info] Environment variables:
[dotnet] 1: [info]   DD_TRACE_ENABLED=
[dotnet] 1: [info]   DD_TRACE_DEBUG=true
[dotnet] 1: [info]   DD_DOTNET_TRACER_HOME=/opt/datadog
[dotnet] 1: [info]   DD_INTEGRATIONS=/opt/datadog/integrations.json
[dotnet] 1: [info]   DD_PROFILER_PROCESSES=
[dotnet] 1: [info]   DD_PROFILER_EXCLUDE_PROCESSES=
[dotnet] 1: [info]   DD_AGENT_HOST=
[dotnet] 1: [info]   DD_TRACE_AGENT_PORT=
[dotnet] 1: [info]   DD_ENV=
[dotnet] 1: [info]   DD_SERVICE_NAME=
[dotnet] 1: [info]   DD_DISABLED_INTEGRATIONS=
[dotnet] 1: [info]   DD_CLR_DISABLE_OPTIMIZATIONS=
[dotnet] 1: [debug] Loading integrations from file: /opt/datadog/integrations.json
[dotnet] 1: [warn] Failed to load integrations: No error information
[dotnet] 1: [warn] Profiler disabled: no enabled integrations found.
```
To answer the obvious, yes /opt/datadog/integrations.json exists.

**Comment by lucaspimentel** [comment]
We are actively working on a build compiled against `musl-libc` to officially support Alpine Linux. See #621.

**Comment by colin-higgins** [comment]
Hello all, we now officially support automatic instrumentation with the new package (datadog-dotnet-apm-1.13.0-musl.tar.gz) available in the 1.13 release (https://github.com/DataDog/dd-trace-dotnet/releases/tag/v1.13.0)
